proposed-api.md

```diff
@@ -0,0 +1,213 @@
+in the static case that can lead to error configurations.  This is discussed below.
+
+## Simple Restrictions
+
+* A vpc may not name any of it's internal IP addresses in more than one `Expose` object
+* A vpc may not map the same external addresses in multiple `Expose` objects
+* A vpc may not have 2 `Route` objects with exactly the same destination and source
+* 2 vpcs may not have a Route using the same external ip addresses that name 
+  IP addresses in the same destination block.
+  * This is because there is no correct way to construct the return routes in these cases.
+  * A single VPC may have multiple routes to overlapping destination IPs with different
+    filter policies, or source IPs
+* 2 `Expose` objects may not expose the same IP addresses, as the gateway has no way
+  to distinguish which IP address was intended as the destination in outbound routes
+
+In the future, we can relax some of the IP addresses in multiple VPC restrictions by having some type of grouping or namespacing of IP blocks.  Then the uniqueness requirements apply only within the group.  Because these VRFs are completely separate the gateway would be able to distinguish which VPC in 2 different groups is the correct VPC and tag it appopropriately to avoid conflicts. Return routes would work the same way.
+
+## Peering objects
+
+To peer multiple VPCs requires the creation of many objects, an `Expose` object for each VPC and a set of `Route` object for each VPC and dest VPC.  To make this easier for users, there is a Peer object that creates these objects for the user:
+
+```
+kind: gateway/Peer
+metadata: 
+  name: peer-vpc1-vpc2
+spec:
+  vpcs:
+    - vpc: vpc1
+      ip: 10.1.1.0/24
+      as: 192.168.10.0/24  # VPC 1's addresses are using NAT
+    - vpc: vp2
+      ip: 10.2.0.0/16
+    - vpc: vpc3
+      ip: 10.3.1.0/24
+```
+
+This would create an `Expose` object for each VPC, and 2 `Route` objects for each VPC which names the other 2 VPCs.
+
+The one issue here is that if these VPCs also `Expose` other addresses, and these are the
+same as the peering addresses, we'll get a conflict.  We should think about what the best
+way to address this is, in a user friendly way.  Perhaps we allow the user to specify peering but ask the Peer object to use a particular `Expose` object instead of generating
+one in some cases.  Then the `Peer` object setup has to check that the `Expose` object
+specified has the correct config not to break peering.
+
+Or perhaps, if there is an existing `Expose` object that conflicts with one that would
+be auto generated, an error is raised if the configuration of the `Expose` object is
+incompatible with the peering, with advice on how to make it compatible.
+
+## Static Routing to Externals
+```
+kind: gateway/Expose
+metadata: 
+  name: expose-external1
+spec:
+  expose:
+    external: "external1"
+    ip: 0.0.0.0/0 # All addresses, no DNAT
```

Comment:
Probably list here as well, for the cases where we only want to expose some other external subnets - e.g. between the fabrics and etc